server/src/main/java/org/elasticsearch/ElasticsearchException.java

```diff
@@ -635,8 +636,25 @@ public static ElasticsearchException failureFromXContent(XContentParser parser)
     public static ElasticsearchException[] guessRootCauses(Throwable t) {
         Throwable ex = ExceptionsHelper.unwrapCause(t);
         if (ex instanceof ElasticsearchException) {
+            // ElasticsearchException knows how to guess its own root cause
             return ((ElasticsearchException) ex).guessRootCauses();
         }
+        if (ex instanceof XContentParseException) {
+            /*
```

Comment:
I feel like if we get *another* exception like this one we should try to build something a little more generic and less "heuristics that make the exception look good". But `XContentParseException isn't really the same thing as `ElasticsearchException`. It is similar, but different enough that I don't think I could boil out the generic bits without this getting wonky.